Tasks/SonarQubePreBuild/SonarQubePreBuildImpl.ps1

```diff
@@ -172,3 +190,95 @@ function IsFilePathSpecified
                 [StringComparison]::OrdinalIgnoreCase)
 }
 
+
+function GetVersion($uri, $headers)
+{
+    $version = $null
+
+    Try
+    {
+        $jsonResp = Invoke-RestMethod -Uri $uri -Method Get -Headers $headers
+
+        if ($jsonResp)
+        {
+            $version = $jsonResp.SonarQube.Version
+        }
+
+    }
+    Catch [System.Net.WebException]
+    {
+        Write-Verbose "WebException while trying to invoke $url. Exception msg:$($_.Exception.Message)"
+    }
+
+    return $version
+}
+
+#
+# Helper that returns the version number of the SonarQube server
+#
+function GetSonarQubeServerVersion()
+{
+    param([String][ValidateNotNullOrEmpty()]$serverUrl,
+          [String][ValidateNotNullOrEmpty()]$userName,
+          [String][ValidateNotNullOrEmpty()]$password)
+
+    Write-Host "Fetching SonarQube server version.."
+
+    $httpHeaders = @{}
+    $serverUri = New-Object -TypeName System.Uri -ArgumentList $serverUrl
+    $serverApiUri = New-Object -TypeName System.Uri -ArgumentList ($serverUri, "/api/system/info")
+
+    $base64Auth = [System.Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes("$($userName):$($password)"))
```

Comment:
For this api call to work, username:password needs to have admin privileges.. check with Bogdan on impact of this requirement.